extensions/README.md

```diff
@@ -1,7 +1,64 @@
 # Jetpack Block Editor Extensions
 
-This directory lists extensions for the Block Editor, also known as Gutenberg,
-[that was introduced in WordPress 5.0](https://wordpress.org/news/2018/12/bebo/).
+This directory contains the source code for extensions in the new post editor,
+also known as Gutenberg, [that was introduced in WordPress 5.0](https://wordpress.org/news/2018/12/bebo/).
+
+## How do I develop block editor extensions in Jetpack?
+
+- Use the [Jetpack Docker environment](https://github.com/Automattic/jetpack/tree/master/docker#readme).
+- Start a new branch.
+- Add your new extension's files to the extensions/blocks directory.
+- Or modify existing extensions in the same folder.
+- Run `yarn build-extensions [--watch]` to compile your changes.
+- Now test your changes in your docker environment's wp-admin.
+- Open a PR, and a WordPress.com diff will be automatically generated with your changes.
+- Test the WordPress.com diff
+- Once the code works well in both environments and has been approved by a Jetpack crew member, you can merge your branch!
+
+## What if I need to manually create a WordPress.com diff?
+You can build extensions from the Jetpack folder to your local sandbox folder and sync the whole sandbox like you always do:
+
+```bash
+yarn build-extensions \
+  --output-path /PATH_TO_YOUR_SANDBOX/wp-content/mu-plugins/jetpack/_inc/blocks/ \
+  --watch
+```
+
+Alternatively, if you don’t need to touch PHP files, you can build extensions in the Jetpack folder without --output-path and use rsync to push files directly to your sandbox:
+
+
+```bash
+rsync -a --delete _inc/blocks/ \
+  YOUR_WPCOM_SANDBOXUSERNAME@YOUR_WPCOM_SANDBOX.wordpress.com:/home/wpcom/public_html/wp-content/mu-plugins/jetpack/_inc/blocks/﻿
+```
+
+Calypso loads Gutenberg from simple sites’ wp-admin in an iframe.
+
+## BETA Extensions
+Generally, all new extensions should start out as beta.
+
+- Before you develop, remember to add your extension's slug to the beta array in `extensions/index.json`.
+- Ensure that, in you docker environment, the `JETPACK_BETA_BLOCKS` constant is set to `true`
+- In the WordPress.com environment, a12s will be able to see beta extensions with no further configuration
+- Once you've successfully beta tested your new extension, you can open new PR to make your extension live!
+- Simply move the extension's slug out of the beta array and into the production array in `extensions/index.json`.
+
+## How do I merge extensions to Jetpack
+- Jetpack is released once a month, so be sure your team is aware of code freeze schedule (PCYsg-eg5-p2).
```

Comment:
Here I would suggest linking to the public page with our milestones instead of the p2 link, since those include release dates:
https://github.com/Automattic/Jetpack/milestones